### PR TITLE
support for ifttt maker channel

### PR DESCRIPTION
**Pull request by khag7**
Note: IFTTT maker channel uses a very simple http request, this code is mostly a copy of the pushover alarm code.

## Description
Added support for IFTTT Maker Channel

## How Has This Been Tested?
Tested on Windows running PokemonGo-Map and using the IFTTT recipe here:
https://ifttt.com/recipes/453460-ifttt-support-test-for-pokealarm

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Wiki Update
In your IFTTT account, connect to the Maker channel. You will be given an API key by IFTTT. Create a new recipe and give it an event name. Make use of the ingredients "value1", "value2", and "value3" in your IFTTT recipe. Note: you cannot change the names of these 3 values, they are set by IFTTT and three is the maximum number available. In PokeAlarm you can edit the config file to enable IFTTT, enter your key and event name, and customize the contents of value1, value2, and value3.

## Checklist:
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by khag7** [comment]
Hi guys. I'm a PHP web developer. Never even used Python until I got into PoGo mapping. Please let me know if I've made any mistakes or whatever. This was extremely simple and works for me, so I think we're good, but make sure to give it a little extra criticism since I'm brand f'n new to python.

File: alarms.json.default [hunk 1]
```diff
@@ -20,7 +20,7 @@
 			"to_number":"YOUR_TO_NUM"
 		},
 		{
-			"active": "True",
+			"active": "False",
```
Comment (khag7) [inline alarms.json.default]: Not sure but this was set to true in the default config file, while all others are false. Pretty sure you want them all to be false.

File: alarms.json.default [hunk 2]
```diff
@@ -38,7 +38,16 @@
 			"access_secret": "YOUR_ACCESS_SECRET",
 			"consumer_key": "YOUR_CONSUMER_KEY",
 			"consumer_secret": "YOUR_CONSUMER_SECRET"
-        }
+        },
+		{
+			"active": "False",
+			"type":"ifttt",
+			"api_key":"bryeq0dWmIgBlDxUiBWaQ3",
```
Comment (khag7) [inline alarms.json.default]: Aaaaand I knew I'd leave this in by mistake. I'll revoke it from my account, I guess you can just fix this before merging?

**Comment by kvangent** [comment]
@khag7 Looks pretty good to me. Few small changes before I merge it:
 
1. I'm in the process of moving the alarms to a more detached style to reduce unessecary component installs. Twitter and Slack are currently in this format if you need an example.
    * Make the import for IFTTT_Alarm be lazy - move it from line 18 to ~54. This way it only imports the alarms if an alarm of that kind is being created. 
    * Move the alarm into it's own module (folder). Make sure to include an init file for it (you can check twitter or slack again, but you won't need to worry about checking for requirements since your using built-ins)

2. Create a mockup of the wiki page in MD - make sure to follow the style of the existing wikis and include the section on how to get an API key

3. Fix the indentation in the Alarms.json :+1: 

I may not be on much over the next day or so, so don't worry if it takes me a while to pull after you update the changes. If you have any questions, let me know.

**Comment by khag7** [comment]
Thanks for the criticism! I appreciate it. Its the weekend and I'm just getting home, maybe tomorrow, maybe the next day, but I'll do it!

**Comment by kvangent** [comment]
@khag7 I've updated the rest of the modules. If you are still interested in getting this merged, please let me know when you have updated yours.

**Comment by khag7** [comment]
Hey sorry I didn't get back to you on this. Been caught up with "real life" things and spending time working on PokemonGo-Map. Hopefully if anyone is interested they can use this as a starting point to get IFTTT support. I actually have been using it. I have some Hue lights which turn on and an alarm plays on my speakers if certain pokemon are within a reasonable range. Also I have the light turn red if the scan stops (i.e. if certain amt of time passes with no webhook sent, I know I need to restart the script). Kind of fun to integrate these things!

**Pull request closed without merging**